include/aws/s3/private/s3_request.h

```diff
@@ -23,6 +24,69 @@ enum aws_s3_request_flags {
     AWS_S3_REQUEST_FLAG_ALWAYS_SEND = 0x00000004,
 };
 
+/**
+ * Information sent in the meta_request telemetry callback.
+ */
+struct aws_s3_request_metrics {
+    struct aws_allocator *allocator;
+
+    struct {
+        /* The time stamp when the request started by S3 client, which is prepared time by the client. Timestamps
+         * are from `aws_high_res_clock_get_ticks`. This will always be available. */
+        uint64_t start_timestamp_ns;
+        /* The time stamp when the request finished by S3 client succeed or failed or to be retried. Timestamps
+         * are from `aws_high_res_clock_get_ticks`. This will always be available. */
+        uint64_t end_timestamp_ns;
+        /* The time duration for the request from start to finish. end_timestamp_ns - start_timestamp_ns. This will
+         * always be available. */
+        uint64_t total_duration_ns;
+
+        /* The time stamp when the request started to be encoded. 0 means data not available. Timestamp
+         * are from `aws_high_res_clock_get_ticks` */
+        uint64_t send_start_timestamp_ns;
+        /* The time stamp when the request finished to be encoded. 0 means data not available.
+         * Timestamp are from `aws_high_res_clock_get_ticks` */
+        uint64_t send_end_timestamp_ns;
+        /* The time duration for the request from start encoding to finish encoding (send_end_timestamp_ns -
+         * send_start_timestamp_ns). When send_end_timestamp_ns is 0, means data not available. */
+        uint64_t sending_duration_ns;
```

Comment:
Did we decide on https://github.com/awslabs/aws-c-s3/pull/271#discussion_r1152314703? I think it would be nice to be consistent with http layer.

Reply:
In my opinion, the reason we used int64_t to have the data unavailable easy to check, but we provide the getters with error, so I guess it will make more sense for those value to be unsigned, as they should never less than zero.